contract/models/res_partner.py

```diff
@@ -7,30 +7,44 @@
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    contract_count = fields.Integer(
-        string='Contracts',
+    sale_contract_count = fields.Integer(
+        string='Sale Contracts',
+        compute='_compute_contract_count',
+    )
+    purchase_contract_count = fields.Integer(
+        string='Purchase Contracts',
         compute='_compute_contract_count',
     )
 
     def _compute_contract_count(self):
-        Contract = self.env['account.analytic.account']
+        contract_model = self.env['account.analytic.account']
         today = fields.Date.today()
+        fetch_data = contract_model.read_group([
+            ('recurring_invoices', '=', True),
```

Comment:
I'm seeing that this doesn't filter by current partners, on the selection, so it's going to be low performance also. You need to get before all the partners (parent and children) for filtering the `read_group` statement.

Reply:
done